### ISSUE TITLE
Gradients through FoM fails when using cubic spline interpolation

### ISSUE THREAD

**Issue by EiffL** [DifferentiableUniverseInitiative/jax_cosmo#57]
For some reason, gradients fail with a cryptic:
```
<class 'jax.ad_util.Zero'> is not a valid JAX type
```
This can be replicated for instance when using this notebook: 
https://github.com/EiffL/tomo_challenge/blob/master/notebooks/LearningToBin-2bins-FoM.ipynb
with the code from this branch: `u/EiffL/spline_interpolation`
